packages/ra-core/src/sideEffect/useUnselectAll.ts

```diff
@@ -10,10 +10,17 @@ import { setListSelectedIds } from '../actions';
  * const unselectAll = useUnselectAll('posts');
  * unselectAll();
  */
-const useUnselectAll = resource1 => {
+const useUnselectAll = (resource1?: string) => {
     const dispatch = useDispatch();
     return useCallback(
-        resource2 => dispatch(setListSelectedIds(resource2 || resource1, [])),
+        (resource2?: string) => {
+            if (!resource2 && !resource1) {
+                console.log(
```

Comment:
maybe you should use the `warning` from `ra-core/util/warning` to hide the warning in production.